lib/assets/javascripts/twine.js.coffee

```diff
@@ -268,10 +268,19 @@ Twine.bindingTypes =
   'bind-class': (node, context, definition) ->
     fn = wrapFunctionString(definition, '$context,$root', node)
     lastValue = {}
+    additions = []
+    removals = []
     return refresh: ->
       newValue = fn.call(node, context, rootContext)
-      for key, value of newValue when !lastValue[key] != !value
-        $(node).toggleClass(key, !!value)
+      for key, value of newValue
+        value = newValue[key] = !!newValue[key]
+        cached = lastValue[key]
+        if cached == undefined and $(node).hasClass(key) != value or cached != value
+          if value then additions.push(key) else removals.push(key)
+      if removals.length
+        $(node).removeClass removals.join(' ')
+      if additions.length
+        $(node).addClass additions.join(' ')
```

Comment:
Either use the postfix notation (`if` after the statement) or add some newlines. Convention in this codebase is also to use brackets for function calls unless the argument spans multiple lines.